Source/MQTTnet/Server/MqttClientSession.cs

```diff
@@ -12,11 +13,16 @@ public class MqttClientSession
 
         private readonly DateTime _createdTimestamp = DateTime.UtcNow;
 
-        public MqttClientSession(string clientId, MqttServerEventDispatcher eventDispatcher, IMqttServerOptions serverOptions, IMqttNetChildLogger logger)
+        public MqttClientSession(MqttConnectPacket connectPacket, MqttServerEventDispatcher eventDispatcher, IMqttServerOptions serverOptions, IMqttNetChildLogger logger)
         {
-            ClientId = clientId ?? throw new ArgumentNullException(nameof(clientId));
+            if (connectPacket == null)
+            {
+                throw new ArgumentNullException(nameof(connectPacket));
+            }
+
+            ClientId = connectPacket.ClientId;
 
-            SubscriptionsManager = new MqttClientSubscriptionsManager(clientId, eventDispatcher, serverOptions);
+            SubscriptionsManager = new MqttClientSubscriptionsManager(connectPacket, eventDispatcher, serverOptions);
             ApplicationMessagesQueue = new MqttClientSessionApplicationMessagesQueue(serverOptions);
```

Comment:
The CONNECT package can change. So the constructor is not enough. See comment below because it covers the same topic but I wrote that comment before this one :smile:.